pyat/at/lattice/utils.py

```diff
@@ -391,7 +394,7 @@ def get_uint32_index(ring: Sequence[Element], refpts: Refpts,
     elif isinstance(refpts, Element):
         checkfun = checktype(type(refpts))
     elif isinstance(refpts, str):
-        checkfun = checkname(refpts)
+        checkfun = checkname(refpts, regex=regex)
```

Comment:
2 remarks here: 
1. I think using `fullmatch` instead to `match` is safer: the pattern must then match the full element name. Otherwise, but matching only the beginning of the name, you may get unexpected matches. It's not penalising: if you are interested only in the first few characters, you can end the pattern with ".*" so that it matches the full element name.
2. The search for matches has to be done on all lattice elements (possibly a lot), so you can gain a lot of time by compiling the RE:
```python
    elif isinstance(refpts, str):
        rgxp = re.compile(pattern)
        return rgxp.fullmatch
```

Reply:
Sorry, my comment is at the wrong place! I wrote it once but it disappeared, so I wrote it again, but in the wrong place!

Of course, this applies to `checkname`, you are right!